cmd/collector/app/span_processor_test.go

```diff
@@ -679,5 +680,5 @@ func TestSpanProcessorWithOnDroppedSpanOption(t *testing.T) {
 	}, processor.SpansOptions{SpanFormat: processor.JaegerSpanFormat})
 
 	assert.NoError(t, err)
-	assert.Equal(t, []string{"op2", "op3"}, droppedOperations)
```

Comment:
I suggest you keep the original test but add an `assert.Eventually` to ensure the right state, like so:

```
assert.NoError(t, err)

assert.Eventually(t, func() bool {
	return len(droppedOperations) == 2
}, 5*time.Second, 10*time.Millisecond)

assert.Equal(t, []string{"op2", "op3"}, droppedOperations)
```